src/formatDate.js

```diff
@@ -50,7 +50,78 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const SEPARATOR_POSITION = 3;
+  const YEARS_LIMIT = 30;
+  const CURRENT_CENTURY = '20';
+  const PREVIOUS_CENTURY = '19';
+  const YEARS = 'YY';
+  const MONTHS = 'MM';
+  const DAYS = 'DD';
+
+  const newSeparator = toFormat[SEPARATOR_POSITION];
+  const oldIndexes = {
+    year: null,
+    month: null,
+    day: null,
+    separator: fromFormat[3],
+  };
+  const newData = [];
+
+  const yearsHandler = (prevFormat, newFormat, value) => {
```

Comment:
Pay attention to naming
![image](https://github.com/mate-academy/js_format_date_advanced/assets/96409563/bf8747aa-03ba-4a53-876f-9875d10de121)